### PR TITLE
add scripts to suggest shard allocation

### PR DESCRIPTION
**Pull request by dannyroberts**

### CODE REVIEW

File: couchdb-cluster-admin/size-shards.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+# usage:
+#   bash couchdb-cluster-admin/size-shards.sh <remote-ip> <remote-data-dir>
+
+ip=$1
+data_dir=$2
+ssh ${ip} "cd ${data_dir}/shards; find . -type f | xargs ls -alS | xargs -L 1 echo | cut -d' ' -f5,9 | sed 's|\./||g'"
```
Comment (emord) [inline couchdb-cluster-admin/size-shards.sh]: Can you assume that all shards are the same size? (I think you can)

If so you can use http://docs.couchdb.org/en/2.0.0/api/database/common.html#get--db to get data/disk size and divide by number of shards. That way it doesn't require ssh commands and it fixes the case when you don't have one server that has all the shards

File: README.md [hunk 2]
```diff
@@ -1,2 +1,14 @@
 # couchdb-cluster-admin
 utility for managing multi-node couchdb 2.x clusters
+
+
+# Help estimating shard allocation
```
Comment (emord) [inline README.md]: When you were looking this up did you see any recommendations on number of shards to use? We may see some improvements by using fewer shards for some indicies or more for synclogs

File: couchdb-cluster-admin/size-shards.sh (hunk 1 above)
Comment (dannyroberts) [reply couchdb-cluster-admin/size-shards.sh]: That's a good point. The shards aren't _exactly_ the same size, but probably close enough to make that worth it. I'll do that in a follow up when I get back to that.

File: README.md (hunk 2 above)
Comment (dannyroberts) [reply README.md]: I didn't look at that at all, no. We're currently on the default of 8 for all the dbs. I think for my current purposes this is totally fine; that kind of tuning may actually make a difference on the likes of ICDS, though.

Comment (emord) [reply README.md]: Yeah. I was specifically thinking of ICDS. though http://docs.couchdb.org/en/2.1.0/cluster/sharding.html#reshard-no-preshard is the only mention I've seen of it (and it doesn't really give you a way to figure out how many to use)

**Pull request merged**